src/config/ConfigBuilder.ts

```diff
@@ -119,8 +119,11 @@ export class ConfigBuilder {
             } else if (key === 'multiplier') {
                 this.effectiveConfig.connectionStrategy.connectionRetry.multiplier = tryGetNumber(value);
             } else if (key === 'clusterConnectTimeoutMillis') {
-                this.effectiveConfig.connectionStrategy.connectionRetry
-                    .clusterConnectTimeoutMillis = tryGetNumber(value);
+                const clusterConnectTimeoutMillis = tryGetNumber(value);
+                if (clusterConnectTimeoutMillis < 0) {
+                    throw new RangeError('clusterConnectTimeoutMillis can only be non-negative');
```

Comment:
No, -1 is a better default value. I meant the error message `clusterConnectTimeoutMillis can be only positive or -1` is wrong, as it can be zero. It should have been `clusterConnectTimeoutMillis can be only non-negative or -1`

Reply:
0 has a special meaning that, do not try to connect to cluster if the attempt fails first time

Reply:
Got your point now